pkg/skaffold/docker/default_repo.go

```diff
@@ -60,11 +59,10 @@ func replace(defaultRepo string, baseImage string) string {
 		if strings.HasPrefix(baseImage, defaultRepo) {
 			return baseImage
 		}
-		// prefixes don't match, concatenate and truncate
-		return truncate(defaultRepo + "/" + baseImage)
+		// prefixes don't match, concatenate image string
+		return defaultRepo + "/" + baseImage
 	}
-
-	return truncate(defaultRepo + "/" + escapeRegex.ReplaceAllString(baseImage, "_"))
+	return defaultRepo + "/" + registry.ESCRegex.ReplaceAllString(baseImage, registry.ReplaceStr)
 }
 
 func truncate(image string) string {
```

Comment:
Add a todo here that the OCI truncation rules apply to the repository name, not the full image name.